sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -904,22 +906,14 @@ object SparkSession {
 
         // No active nor global default session. Create a new one.
         val sparkContext = userSuppliedContext.getOrElse {
-          // set app name if not given
-          val randomAppName = java.util.UUID.randomUUID().toString
           val sparkConf = new SparkConf()
           options.foreach { case (k, v) => sparkConf.set(k, v) }
-          if (!sparkConf.contains("spark.app.name")) {
-            sparkConf.setAppName(randomAppName)
-          }
-          val sc = SparkContext.getOrCreate(sparkConf)
-          // maybe this is an existing SparkContext, update its SparkConf which maybe used
-          // by SparkSession
-          options.foreach { case (k, v) => sc.conf.set(k, v) }
-          if (!sc.conf.contains("spark.app.name")) {
-            sc.conf.setAppName(randomAppName)
-          }
-          sc
+          // set a random app name if not given.
+          sparkConf.setAppName(options.getOrElse("spark.app.name",
+            java.util.UUID.randomUUID().toString))
+          SparkContext.getOrCreate(sparkConf)
         }
+        options.foreach { case (k, v) => sparkContext.conf.set(k, v) }
```

Comment:
We already did this in https://github.com/apache/spark/pull/18512/files#diff-d91c284798f1c98bf03a31855e26d71cL917 .

The only difference is about `userSuppliedContext`, which is given by a [private method](https://github.com/apache/spark/pull/18512/files#diff-d91c284798f1c98bf03a31855e26d71cR767), so we should be able to check if it's safe or not.